### PR TITLE
fixes #1650: Updated the executability visitor

### LINKED ISSUE #1650: Full Table scan errors need to provide a reason why
When a query is deemed non-executable by the executability visitor, then the user will get a "full table scan required" message which does not help in determining how to modify the query.  Given the fix for #1558  fixed in #1566 , this may be occurring more often especially when model expansions are involved.

### PR DESCRIPTION
**Pull request by ivakegg**
* executability visitor provides a better query debug output
* Also provides a summary of the reason for a executability state
* Query planner will return the summary with the exception

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java [hunk 1]
```diff
@@ -369,51 +516,24 @@ else if (states.contains(STATE.EXECUTABLE)) {
...
     }
     
     @Override
     public Object visit(ASTEQNode node, Object data) {
+        if (output != null) {
+            output.disable();
+        }
+        
         STATE state;
+        Set<STATE> states = new HashSet<>();
```
Comment (jwomeara) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java]: Where is this used?  Looks like it isn't.

**Review by jwomeara — CHANGES_REQUESTED** [review]
Overall, looks good.  Some comments for the Output class and methods, plus some test cases would be appreciated.

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java (hunk 1 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java]: will remove

**Review by jwomeara — DISMISSED** [review]

**Review by apmoriarty — APPROVED** [review]
LGTM

**Review by FineAndDandy — APPROVED** [review]

**Pull request merged**